Add for persistence Chaos Dashboard (#267)
chaos-mesh/website: docs/persistence-dashboard.md

---
title: Persistence Chaos Dashboard
---

import PickHelmVersion from '@site/src/components/PickHelmVersion'

This document describes how to persistence Chaos Dashboard.

Chaos Dashboard support `SQLite`, `MySQL` and `Postgres` as database backend for persistence.

> ## SQLite (Default Database Engine)

Reviewer: ```suggestion
## SQLite (default)
```

    existingClaim: ""

    # Chaos Dashboard data Persistent Volume size.
    size: 8Gi

    # Chaos Dashboard data Persistent Volume Storage Class.
    # If defined, storageClassName: <storageClass>
    storageClassName: standard

    # Chaos Dashboard data Persistent Volume mount root path
    mountPath: /data

    # Subdirectory of Chaos Dashboard data Persistent Volume to mount
    # Useful if the volume's root directory is not empty
    subPath: ""
```

:::warning

If Chaos Dashboard component restarts without `PV`, the data of Chaos Dashboard will be list and can't be retrieved.

:::

## MySQL

Chaos Dashboard supports MySQL 5.6 and higher as the database engine. The below example demonstrates MySQL database configuration. Please reference the official driver [documentation](https://github.com/go-sql-driver/mysql#dsn-data-source-name) for connection string configuration details.

<PickHelmVersion>
helm install chaos-mesh chaos-mesh/chaos-mesh -n=chaos-testing --version latest --set dashboard.env.DATABASE_DRIVER=mysql --set dashboard.env.DATABASE_DATASOURCE=root:password@tcp(1.2.3.4:3306)/chaos-mesh?parseTime=true
</PickHelmVersion>

> ## Postgres

Reviewer: ```suggestion
## PostgreSQL
```

> Chaos Dashboard supports Postgres 9.6 and higher as the database engine. The below example demonstrates Postgres database configuration. Please reference the official driver [documentation](https://www.postgresql.org/docs/current/static/libpq-connect.html#LIBPQ-CONNSTRING) for connection string configuration details.

Reviewer: ```suggestion
Chaos Dashboard supports PostgreSQL 9.6 and higher versions as the database engine. The below example demonstrates PostgreSQL database configuration. For details about connection string configuration, refer to [libpq connect](https://www.postgresql.org/docs/current/static/libpq-connect.html#LIBPQ-CONNSTRING).
```

<PickHelmVersion>
helm install chaos-mesh chaos-mesh/chaos-mesh -n=chaos-testing --version latest --set dashboard.env.DATABASE_DRIVER=postgres --set dashboard.env.DATABASE_DATASOURCE=postgres://root:password@1.2.3.4:5432/postgres?sslmode=disable
</PickHelmVersion>

> ## Set TTL(Time To Live) for Chaos Dashboard Data

Reviewer: ```suggestion
## Set TTL (Time To Live) for Chaos Dashboard data
```

> Chaos Dashboard supports setting the expiration time of Chaos Dashboard data. The default `Event` related data expires by `168h`, and the `Experiment` related data defaults to `336h`. If you need to modify it, you can set `dashboard.env.TTL_EVENT` and `dashboard.env` .TTL_EXPERIMENT` parameter, like:

Reviewer: ```suggestion
Chaos Dashboard supports setting the expiration time of Chaos Dashboard data. The default `Event` related data expires by `168h`, and the `Experiment` related data defaults to `336h`. If you need to modify it, you can set `dashboard.env.TTL_EVENT` and `dashboard.env.TTL_EXPERIMENT` parameters, like:
```
